scene/animation/animation_blend_tree.cpp

```diff
@@ -245,6 +245,8 @@ AnimationNode::NodeTimeInfo AnimationNodeAnimation::_process(const AnimationMixe
 
 	if (!p_test_only) {
 		AnimationMixer::PlaybackInfo pi = p_playback_info;
+		pi.start = start_offset;
```

Comment:
This almost works, but it still breaks root motion if you're using an animation with a custom timeline. I still recommend changing it to this which seems to fix the bug:
```
		pi.start = 0.0;
		pi.end = cur_len;
```